src/libraries/System.Linq.AsyncEnumerable/src/System/Linq/Cast.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Linq
+{
+    public static partial class AsyncEnumerable
+    {
+        // TODO: Consider before shipping .NET 10 whether this can use extension everything to improve generic inference.
```

Comment:
Should we replace this TODO with an issue?

Reply:
Yes, I will open one before merging.